### PR TITLE
Adding `object` hit based tests and test assets - preparing for PHP 7.2 compatibility

### PR DESCRIPTION
**Pull request by Ocramius**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
         }
     ],
     "require": {
-        "php":                       "^7.1.0",
+        "php":                       "^7.2.0",
```
Comment (Majkl578) [inline composer.json]: You should also change Travis to run builds under 7.2 - it's supported as of today.

Comment (Ocramius) [inline composer.json]: Oh, nice! I thought `nightly` was all there was to it :)

Comment (Majkl578) [inline composer.json]: `nightly` is now 7.3.0-dev. :)

**Comment by Ocramius** [comment]
Aaaand tests pass \o/

Just can't get humbug to run yet, because it needs xdebug :-\

**Comment by Majkl578** [comment]
Also PHPStan fails due to unsupported `object` type - upstream refuses to fix it before 7.2 GA. :( https://github.com/phpstan/phpstan/issues/367

**Comment by Ocramius** [comment]
This requires phpstan `0.9.x` as per https://github.com/phpstan/phpstan/issues/367#issuecomment-329612065

**Comment by Ocramius** [comment]
:ship:

**Pull request merged**